lib/pengine/bundle.c

```diff
@@ -2030,22 +2035,16 @@ free_bundle_replica(pcmk__bundle_replica_t *replica)
     }
 
     if (replica->ip) {
-        free_xml(replica->ip->xml);
-        replica->ip->xml = NULL;
+        free_xml(replica->ip->private->xml);
```

Comment:
Pretty sure this is still safe/necessary with the current implementation (at least as long as there's no template involved). The `free()` method calls `common_free()`, which frees `rsc->private->xml` if the resource is marked as removed with no parent or if the XML was expanded from a template. Of course in this case there is a parent.

We may want to continue setting `replica->ip->private->xml` to `NULL` before calling `free()`, for future-proofing. No strong feelings. We still do so in `pe__free_bundle()`.

Same for others below.

Reply:
good catch